src/DefaultTokenUriResolver.sol

```diff
@@ -0,0 +1,590 @@
+
+    constructor(
+        IJBOperatorStore _operatorStore,
+        IJBDirectory _directory,
+        IJBProjectHandles _projectHandles,
+        ITypeface _capsulesTypeface
+        // IReverseRegistrar _reverseRegistrar,
+        // IResolver _resolver
+    ) JBOperatable(_operatorStore) {
+        directory = _directory;
+        projects = directory.projects();
+        fundingCycleStore = directory.fundingCycleStore();
+        controller = IJBController(directory.controllerOf(1));
+        tokenStore = controller.tokenStore();
+        singleTokenPaymentTerminalStore = IJBSingleTokenPaymentTerminalStore(IJBPayoutRedemptionPaymentTerminal(address(IJBPaymentTerminal(directory.primaryTerminalOf(1, JBTokens.ETH)))).store());
+        projectHandles = _projectHandles;
+        capsulesTypeface = _capsulesTypeface;
+        // reverseRegistrar = _reverseRegistrar;
+        // resolver = _resolver;
+        themes[0] = Theme({
+            projectId: 0,
+            textColor: "#FF9213",
+            bgColor: "#44190F",
+            bgColorDark: "#3A0F0C"
+        });
+    }
+
+    // @notice Gets the Base64 encoded Capsules-500.otf typeface
+    /// @return fontSource The Base64 encoded font file
+    function getFontSource() internal view returns (bytes memory fontSource) {
+        return
+            ITypeface(capsulesTypeface).sourceOf(
+                Font({weight: 500, style: "normal"})
+            ); // Capsules font source
+    }
+
+    /// @notice Transform strings to target length by abbreviation or left padding with spaces.
+    /// @dev Shortens long strings to 13 characters including an ellipsis and adds left padding spaces to short strings. Allows variable target length to account for strings that have unicode characters that are longer than 1 byte but only take up 1 character space.
+    /// @param left True adds padding to the left of the passed string, and false adds padding to the right
+    /// @param str The string to transform
+    /// @param targetLength The length of the string to return
+    /// @return string The transformed string
+    function pad(
+        bool left,
+        string memory str,
+        uint256 targetLength
+    ) internal view returns (string memory) {
+        uint256 length = bytes(str).length;
+        if (left) {
+            // Left pad
+            if (length > targetLength) {
+                // Shorten strings strings longer than target length
+                str = string.concat(
+                    slice.slice(str, 0, targetLength - 1),
+                    unicode"…"
+                ); // Shortens to 1 character less than target length and adds an ellipsis unicode character
+            } else if (length == targetLength) {
+                return str;
+            } else {
+                // Pad strings shorter than target length
+                string memory padding;
+                for (uint256 i = 0; i < targetLength - length; i++) {
```

Comment:
`unchecked` to save gas on loop increments

Reply:
```suggestion
                uint256 _numberOfChar = targetLength - length
                for (uint256 i; i < _numberOfChar; i++) {
```

Reply:
Given that it's only used in a View fn, I'm just gonna go with Dr.G's clarifying suggestion and leave the Unchecked suggestion out. Can revisit in future upgrades but the SVG is so heavy I don't anticipate anyone composing ON TOP of it even in a view fn.

Reply:
Resolved by viraz [71a4dac](https://github.com/jbx-protocol/juice-token-resolver/pull/1/commits/71a4dace79243a883f3df342e2a711ffdddb50d4)